src/contexts/Connect/index.tsx

```diff
@@ -368,6 +499,103 @@ export const ConnectProvider = ({
       updateInitialisedExtensions(
         `unknown_extension_${extensionsInitialisedRef.current.length + 1}`
       );
+    } else if (id === 'wallet-connect') {
+      const provider = await WalletConnectInitializer.initialize();
```

Comment:
@marshacb Do we only need to do this if there is not an existing provider assigned to `wcChainInfo`?

Reply:
@rossbulat yes, if there is already a provider/session assigned in the state we don't need to re-initialize one again

Reply:
Cool okay,I'll explore when we can cut down on code. 👍